Update to Lodestone 0.9.0 and Gradle 8.1.1

SizableShrimp opened the pull request:
This PR is waiting on the merge of ParchmentMC/Lodestone#1, assuming it is tagged as `v0.3.0` when merged.

* Lodestone has been updated to `0.3.0`.
* The new `lodestone` extension object is in use to set the minecraft version.
* Tasks have been cleaned up to use mapped task providers as inputs, removing the need to explicitly declare `dependsOn` for most tasks.
* A few tasks have been renamed to better match what they do, and all defined variables now exactly match their task names.
* Publishing now uses the resolved minecraft version.

SizableShrimp committed:
Update to Lodestone 0.3.0 and Gradle 7.1.1

SizableShrimp committed:
Print resolved mc version to gradle console

sciwhiz12 committed:
Bump updated Lodestone to v0.3.1

Done upon request of @SizableShrimp.

marchermans commented:
This pr is completely out of date and completely not needed.

Lodestone is primarily a task provider and not a tool configuration system. I do not want to use the extension object since i personally consider it to be the inferior design system to just declaring my tasks and my task dependency chain, what is wrong with that?

marchermans reviewed (changes requested):
See comment above on the state of this pr and it's necessity

sciwhiz12 commented:
First, the use of the `lodestone` extension reduces the task configuration needed in this repository's buildscript, as the configuration of the MC version in the plugin's tasks are handled by the plugin itself through the `lodestone` extension.

Second, the use of the `lodestone` extension is supposed to resolve #2, because the tasks now use the resolved Minecraft version rather than the input version directly, which may be `latest` or `latest-snapshot`. Currently, the GH issue is closed, but the actual issue remains unresolved while this PR remains unmerged. I assume you were aware of this when you merged the original Lodestone PR which added the extension...?

I consider the extension object to be a design that more follows the Gradle-recommended principle of 'convention over configuration'. It allows moving the tedium of task setup from individual buildscripts to the plugin while leaving configuration to the user, while still allowing individual users to make their own task tree as needed.

sciwhiz12 committed:
Update Gradle to 8.1.1

sciwhiz12 committed:
Remove MaxPermSize from JVM arguments

This VM option was removed in later versions than Java 8, which means
Java 17 (which is also the current version on the CI's agents) throws
an error and refuses to start.

sciwhiz12 committed:
Update Lodestone to 0.9.0

sciwhiz12 committed:
Merge branch 'main'
